File: ProjetComparatif/python/tache2_tri_recherche.py
import pandas as pd
import time

start = time.time()

# Charger les données
df = pd.read_csv("notes_etudiants.csv")

# Extraire et trier les noms par ordre alphabétique
noms_tries = sorted(df["Nom"])

# Rechercher les noms terminant par "e"
noms_finissant_par_e = [nom for nom in noms_tries if nom.endswith("e")]

# Affichage des résultats
print("Étudiants dont le nom se termine par 'e' :\n")
for nom in noms_finissant_par_e:
    print(nom)

# Temps d'exécution
end = time.time()
print(f"\n⏱ Temps d'exécution : {(end - start) * 1000:.3f} millisecondes")

File: ProjetComparatif/Csharp/Tache2/Tache2_TriRecherche.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.IO;
using System.Linq;

class Tache2_TriRecherche
{
    static void Main()
    {
        Stopwatch stopwatch = new Stopwatch();
        stopwatch.Start();

        string filePath = "notes_etudiants.csv";

        if (!File.Exists(filePath))
        {
            Console.WriteLine($"Fichier introuvable : {filePath}");
            return;
        }

        var lignes = File.ReadAllLines(filePath).Skip(1); // Sauter l'en-tête
        var noms = new List<string>();

        foreach (var ligne in lignes)
        {
            var colonnes = ligne.Split(',');
            if (colonnes.Length > 0)
                noms.Add(colonnes[0]); // Supposer que le nom est dans la première colonne
        }

        // Trier les noms
        noms.Sort();

        // Rechercher ceux qui se terminent par "e"
        var nomsFinissantParE = noms.Where(n => n.EndsWith("e")).ToList();

        // Affichage des résultats
        Console.WriteLine("Étudiants dont le nom se termine par 'e' :\n");
        foreach (var nom in nomsFinissantParE)
        {
            Console.WriteLine(nom);
        }

        stopwatch.Stop();
        Console.WriteLine($"\n⏱ Temps d'exécution : {stopwatch.Elapsed.TotalSeconds:F6} secondes");
    }
}
